### PR TITLE
HIVE-26770: Make end of loop compaction logs appear more selectively and reduce code duplication

### PR DESCRIPTION
**Pull request by akshat0395**
### What changes were proposed in this pull request?

Make "end of loop" compaction logs appear more selectively and move duplicate code from Compactor threads to base class, more details can be found in the following ticket
[HIVE-26770](https://issues.apache.org/jira/browse/HIVE-26770)

### Why are the changes needed?

Improved logging for Compactor threads to reduce noise and share time based stats
### Does this PR introduce _any_ user-facing change?

No

### How was this patch tested?

Unit tests

### CODE REVIEW

File: ql/src/java/org/apache/hadoop/hive/ql/txn/compactor/Worker.java [hunk 1]
```diff
@@ -138,6 +142,7 @@ public void run() {
   @Override
   public void init(AtomicBoolean stop) throws Exception {
     super.init(stop);
+    checkInterval = 0;
```
Comment (deniskuzZ) [inline ql/src/java/org/apache/hadoop/hive/ql/txn/compactor/Worker.java:145]: set it to 0 in the declaration

File: ql/src/java/org/apache/hadoop/hive/ql/txn/compactor/CompactorThread.java [hunk 2]
```diff
@@ -61,6 +61,13 @@ public abstract class CompactorThread extends Thread implements Configurable {
...
   protected String runtimeVersion;
 
+  //Time threshold for compactor thread log
+  //In milliseconds:
+  protected Integer MAX_WARN_LOG_TIME = 1200000; //20 min
+
+  protected long checkInterval;
+
+  public enum CompactorThreadType {INITIATOR, WORKER, CLEANER}
   @Override
```
Comment (deniskuzZ) [inline ql/src/java/org/apache/hadoop/hive/ql/txn/compactor/CompactorThread.java:71]: new line + should it be public or package-private is enough?

File: ql/src/java/org/apache/hadoop/hive/ql/txn/compactor/CompactorThread.java [hunk 3]
```diff
@@ -61,6 +61,14 @@ public abstract class CompactorThread extends Thread implements Configurable {
   protected String hostName;
   protected String runtimeVersion;
 
+  //Time threshold for compactor thread log
+  //In milliseconds:
+  protected Integer MAX_WARN_LOG_TIME = 1200000; //20 min
+
+  protected long checkInterval = 0;
+
+  protected enum CompactorThreadType {INITIATOR, WORKER, CLEANER}
```
Comment (deniskuzZ) [inline ql/src/java/org/apache/hadoop/hive/ql/txn/compactor/CompactorThread.java:70]: minor: can't we use package-private visibility?

File: ql/src/java/org/apache/hadoop/hive/ql/txn/compactor/CompactorThread.java [hunk 4]
```diff
@@ -61,6 +61,14 @@ public abstract class CompactorThread extends Thread implements Configurable {
   protected String hostName;
   protected String runtimeVersion;
 
+  //Time threshold for compactor thread log
+  //In milliseconds:
+  protected Integer MAX_WARN_LOG_TIME = 1200000; //20 min
```
Comment (deniskuzZ) [inline ql/src/java/org/apache/hadoop/hive/ql/txn/compactor/CompactorThread.java:66]: reported in checkstyle: should be marked as static final (constant)

File: ql/src/java/org/apache/hadoop/hive/ql/txn/compactor/CompactorThread.java (hunk 3 above)
Comment (akshat0395) [reply ql/src/java/org/apache/hadoop/hive/ql/txn/compactor/CompactorThread.java:70]: Initial thought for this was to be used by just the subclasses, but on a second thought makes sense for it to be package-private for other classes in package to use this as well in future. Will change

**Review by deniskuzZ — APPROVED** [review]
LGTM +1

**Pull request merged**